File: Code/server/app/models/Pet.py
from enum import Enum
from datetime import datetime
from typing import Optional
from .User import User
from odmantic import Model, Reference, Field


class Gender(str, Enum):
    MALE = "Male"
    FEMALE = "Female"


class Pet(Model):
    avatar: Optional[str]
    name: str = Field(max_length=50)
    birthday: datetime
    gender: Gender
    type: str
    breed: Optional[str]
    want_marraige: bool = True
    owner: User = Reference()
    likes: int = 0
    followers: int = 0
    created_at: datetime = Field(default_factory=datetime.utcnow)
    

    class Config:
        collection = 'pets'
        parse_doc_with_default_factories = True

class Follower(Model):
    following: Pet = Reference()
    follower: User = Reference()
    created_at: datetime = Field(default_factory=datetime.utcnow)

    class Config:
        collection = 'followers'
        parse_doc_with_default_factories = True

File: Code/petiapp/lib/models/pet.dart
import 'user.dart';

class Pet {
  Pet(
      {this.avatar,
      this.name,
      this.birthday,
      this.gender,
      this.type,
      this.breed,
      this.wantMarraige,
      this.owner,
      this.createdAt,
      this.likes,
      this.followers,
      this.id});

  String? avatar;
  String? name;
  String? birthday;
  String? gender;
  String? type;
  String? breed;
  bool? wantMarraige;
  User? owner;
  String? createdAt;
  int? likes;
  int? followers;
  String? id;

  factory Pet.fromMap(Map<String, dynamic> json) => Pet(
        avatar: json["avatar"],
        name: json["name"],
        birthday: json["birthday"],
        gender: json["gender"],
        type: json["type"],
        likes: json["likes"],
        followers: json["followers"],
        breed: json["breed"] == null ? null : json["breeds"],
        wantMarraige: json["want_marraige"],
        owner: json["owner"] == null
            ? json["owner_id"] == null
                ? null
                : User(id: json["owner_id"])
            : User.fromMap(json["owner"]),
        createdAt: json["created_at"],
        id: json["id"],
      );

  String get birthDate => birthday!.split('T')[0];

  String get birthText => 'Birthday: $birthDate';
}
